src/Features/VisualBasic/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.vb

```diff
@@ -0,0 +1,501 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.Composition
+Imports System.Threading
+Imports Microsoft.CodeAnalysis.CodeActions
+Imports Microsoft.CodeAnalysis.CodeRefactorings
+Imports Microsoft.CodeAnalysis.Formatting
+Imports Microsoft.CodeAnalysis.Simplification
+Imports Microsoft.CodeAnalysis.Text
+Imports Microsoft.CodeAnalysis.VisualBasic
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.CodeRefactorings.InvertIf
+    <ExportCodeRefactoringProvider(LanguageNames.VisualBasic, Name:=PredefinedCodeRefactoringProviderNames.InvertIf), [Shared]>
+    Friend Class InvertIfCodeRefactoringProvider
+        Inherits CodeRefactoringProvider
```

Comment:
i'm surprised by the amount of duplicatoin between this and hte C# version. I would have expected a common base class that shared a bunch of the "high level" concepts here, with calls to the subclasses to do the fidly bits where C# and VB differ substantively

Reply:
@CyrusNajmabadi , agreed.  But since this was written several years ago and seems to be working well as is, we're opting to not rewrite it at this point.